backend/src/apiserver/server/pipeline_server.go

```diff
@@ -243,6 +310,24 @@ func (s *PipelineServer) CreatePipelineVersion(ctx context.Context, request *api
 		return nil, util.Wrap(err, "The URL is valid but pipeline system failed to read the file.")
 	}
 
+	// Extract pipeline id and authorize
+	var pipelineId = ""
+	for _, resourceReference := range request.Version.ResourceReferences {
+		if resourceReference.Key.Type == api.ResourceType_PIPELINE && resourceReference.Relationship == api.Relationship_OWNER {
+			pipelineId = resourceReference.Key.Id
+		}
+	}
+	if len(pipelineId) == 0 {
+		return nil, util.Wrap(err, "Create pipeline version failed due to missing pipeline id")
+	}
+	resourceAttributes := &authorizationv1.ResourceAttributes{
+		Verb: common.RbacResourceVerbList,
+	}
+	err = s.CanAccessPipeline(ctx, pipelineId, resourceAttributes)
```

Comment:
similar above, we can write the error check via this
```
if err := doStuff(); err != nil {
  // handle the error here
}
```

Reply:
done